include/swift/AST/FineGrainedDependencyFormat.h

```diff
@@ -0,0 +1,106 @@
+//===---- FineGrainedDependencyFormat.h - swiftdeps format ---*- C++ -*-======//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_AST_FINEGRAINEDDEPENDENCYFORMAT_H
+#define SWIFT_AST_FINEGRAINEDDEPENDENCYFORMAT_H
+
+#include "llvm/Bitcode/RecordLayout.h"
+#include "llvm/Bitstream/BitCodes.h"
+
+namespace llvm {
+class MemoryBuffer;
+}
+
+namespace swift {
+
+class DiagnosticEngine;
+
+namespace fine_grained_dependencies {
+
+class SourceFileDepGraph;
+
+using llvm::BCFixed;
+using llvm::BCVBR;
+using llvm::BCBlob;
+using llvm::BCRecordLayout;
+
+const unsigned char FINE_GRAINED_DEPDENENCY_FORMAT_SIGNATURE[] = {'D', 'E', 'P', 'S'};
+
+const unsigned FINE_GRAINED_DEPENDENCY_FORMAT_VERSION_MAJOR = 1;
+
+/// Increment this on every change.
+const unsigned FINE_GRAINED_DEPENDENCY_FORMAT_VERSION_MINOR = 0;
+
+using IdentifierIDField = BCVBR<13>;
```

Comment:
`BCVBR` is a variable-width encoding. `13` is the number of bits plus one which are encoded per chunk. I arrived at 2**12 empirically, but I don't think it matters too much.

Reply:
What would happen if the value ever exceeded that? Would it just be slower, or would it break? If the latter, what would the error message be? Could a bug somewhere cause the code to *think* the value were more than 2**12 and would the result be reasonable to debug?

If the 13 is because the code counts on there being less than 2**12 names in a file, a comment would be nice.